lib/parser.js

```diff
@@ -613,7 +613,15 @@ let whileLoop = seqMap(keywords.while,
 let ifElse = seqMap(keywords.if,
   expr,
   block,
-  (keywords.else).then(block).times(0, 1).map((elses) => {
+  (keywords.else).then(alt(block, lazy(() => {
+    return ifElse;
+  }).source().map((elseIf) =>({
```

Comment:
I think it'd be clearer if the entirety of ifElse was nested in a call to lazy(). Then you can omit the lazy() here.